src/builtins/BuiltinSharedArrayBuffer.cpp

```diff
@@ -40,27 +40,58 @@ static Value builtinSharedArrayBufferConstructor(ExecutionState& state, Value th
         ErrorObject::throwBuiltinError(state, ErrorObject::RangeError, state.context()->staticStrings().SharedArrayBuffer.string(), false, String::emptyString, ErrorObject::Messages::GlobalObject_FirstArgumentInvalidLength);
     }
 
-    return SharedArrayBufferObject::allocateSharedArrayBuffer(state, newTarget.value(), byteLength);
+    Optional<uint64_t> maxByteLength;
+    // Let requestedMaxByteLength be ? GetArrayBufferMaxByteLengthOption(options).
+    if (UNLIKELY((argc > 1) && argv[1].isObject())) {
+        Object* options = argv[1].asObject();
+        Value maxLengthValue = options->get(state, ObjectPropertyName(state.context()->staticStrings().maxByteLength)).value(state, options);
+        if (!maxLengthValue.isUndefined()) {
+            maxByteLength = maxLengthValue.toIndex(state);
+            if (UNLIKELY((maxByteLength.value() == Value::InvalidIndexValue) || (byteLength > maxByteLength.value()))) {
```

Comment:
In both cases, should the user be informed `FirstArgumentInvalidLength`?

Reply:
Its my mistake. I'll fix it with proper error message.

Reply:
Fixed it. Thanks for the review!